cas-server-webapp-support/src/main/java/org/jasig/cas/report/sso/DefaultSingleSignOnSessionsReport.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.cas.report.sso;
+
+import org.apache.commons.collections.Predicate;
+import org.jasig.cas.CentralAuthenticationService;
+import org.jasig.cas.ticket.Ticket;
+import org.jasig.cas.ticket.TicketGrantingTicket;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Default implementation of the {@link SingleSignOnSessionsReport}.
+ * @author Dmitriy Kopylenko
+ * @author Misagh Moayyed
+ * @since 4.1
+ */
+@Component
+public class DefaultSingleSignOnSessionsReport implements SingleSignOnSessionsReport {
```

Comment:
Point taken, but given that we are modifying the statistics features of CAS, this seemed like a natural fit to produce more stats. Does that make sense, or would you still rather see this in a different PR?